### PR TITLE
Change Callable implementation

### PR DESCRIPTION
**Pull request by shravan97**
Features in this PR : 
* Change the implementation of the callable function to return only the values of the executed action nodes
* Add a new function to get the list of action nodes in the same order as they were executed 
* Modify tests to incur the implementation changes

### CODE REVIEW

File: PyRDF/backend/Local.py [hunk 1]
```diff
@@ -51,7 +51,9 @@ def execute(self, generator):
 
         rdf = ROOT.ROOT.RDataFrame(*generator.head_node.args) # Create RDF object
 
-        values, nodes = mapper(rdf) # Execute the mapper function
+        values = mapper(rdf) # Execute the mapper function
+        nodes = generator.get_action_nodes()
+        # Get the action nodes in the same order as values
```
Comment (etejedor) [inline PyRDF/backend/Local.py]: Should this comment precede (or appear next to) the previous line?

File: PyRDF/CallableGenerator.py [hunk 2]
```diff
@@ -21,6 +21,45 @@ def __init__(self, head_node):
...
+
+        """
+        return_nodes = []
+
+        if not node_py:
+            # In the first recursive state, just set the
+            # current PyRDF node as the head node
+            node_py = self.head_node
+        else:
+
```
Comment (etejedor) [inline PyRDF/CallableGenerator.py]: Remove extra newline?

File: PyRDF/backend/Local.py (hunk 1 above)
Comment (shravan97) [reply PyRDF/backend/Local.py]: It should appear on line 56 after `nodes = generator.get_action_nodes()`. I put it below as that line would get too long in that case

File: PyRDF/CallableGenerator.py (hunk 2 above)
Comment (shravan97) [reply PyRDF/CallableGenerator.py]: I'll remove this @etejedor 👍

File: PyRDF/CallableGenerator.py [hunk 3]
```diff
@@ -21,6 +21,45 @@ def __init__(self, head_node):
...
+        ----------
+        node_py (optional)
+            The current state's PyRDF node. If `None`,
+            it takes the value of `self.head_node`.
+
+        Returns
+        -------
+            list
+                A list of action nodes in the same order as the
+                order of execution in the callable function.
```
Comment (etejedor) [inline PyRDF/CallableGenerator.py]: Can we say: A list of the action nodes of the graph in DFS order, which coincides with the order of execution in the callable function.

Comment (shravan97) [reply PyRDF/CallableGenerator.py]: Yup @etejedor . That sounds better. I'll modify it 👍

File: PyRDF/CallableGenerator.py [hunk 4]
```diff
@@ -49,20 +88,21 @@ def mapper(node_cpp, node_py=None):
...
+
             Returns
             -------
-                tuple
-                    This tuple consists of 2 lists, the first one
-                    consisting of output values of action nodes and
-                    the second one of action nodes in the same order
-                    as that of the values list.
+                list
+                    This is a list of values in the order that they
```
Comment (etejedor) [inline PyRDF/CallableGenerator.py]: And here I would say: "A list of `TResultPtr` objects in DFS order of their corresponding actions in the graph."

@shravan97 , note that here we do not execute anything, the whole event loop will be triggered when accessing one of the `TResultPtr`s later in the backend.

**Comment by etejedor** [comment]
Once the comments I made are applied, I think this is good to go. It is a pity that we have some code repetition because of the double traversal of the graph, but we need to do it this way because the distributed backends will only get the corresponding nodes at the very end, in the client. Having the values of the nodes in the workers is useless.

File: PyRDF/backend/Local.py (hunk 1 above)
Comment (etejedor) [reply PyRDF/backend/Local.py]: Then I would put the comment in the line above.

File: PyRDF/CallableGenerator.py (hunk 4 above)
Comment (shravan97) [reply PyRDF/CallableGenerator.py]: Yup @etejedor . Will change this 👍

File: PyRDF/backend/Local.py (hunk 1 above)
Comment (shravan97) [reply PyRDF/backend/Local.py]: I'll make the change @etejedor  👍

**Comment by shravan97** [comment]
Yes @etejedor . I agree. I believe there's just a difference of 1 line between `get_action_nodes` and `mapper`

**Comment by shravan97** [comment]
@etejedor done now 👍

**Comment by shravan97** [comment]
@etejedor and @diocas , rebase done 👍

**Pull request merged**